### DISCUSSION TITLE
Few suggestions to simplify things

### DISCUSSION THREAD

**Discussion by chan-dev** [stephanrauh/ngx-extended-pdf-viewer#920] [Q&A]
Just take note that I may have missed some of the things in the documentation.

Based on what I've been experiencing right now, there are few things that I think could simplify things:

1. Allow an option to disable `file` listening under the hood, so users can make their custom logic when listening to `file` query. 
Let's say you want to call a external API first without going thru all the checking inside pdf.js
Because right now, the situation is let's say i have this url
`http://www.test.com?file="http://some-api/pdfs/idOfPdf"`, what happens here is the url from `file` query will undergo checking under the hood in pdf.js before being processed by your own code.
2. Allow to link the path of a custom pdf.js (if exclusively just editing pdf.js; this would relieve anyone from manually compiling multiple repos). 
3. Allow bubbling up of pdf.js errors for custom error handling. Something like `(errors)="handleError($error)`
 Right now, in my case, even if the API call i have in my app is successful, it will return a error inside the internal checking of pdf.js.
 Let's say i pass a url from the previous example: `http://www.test.com?file="http://some-api/pdfs/idOfPdf"` but i want to process this url first to attach a `?token=tokenValue` query parameter
 What would happen behind the scenes is `http://some-api/pdfs/idOfPdf` will be requested by pdf.js behind the scenes even before i process this value and will return an error because it has no token yet.

**Comment by stephanrauh** [comment]
Ad 1: I prefer to remove the `?file=` listener entirely. I've opened a bug ticket (#926).

Ad 2: You can achieve the this by adding your copy of pdf.js and - if necessary - pdf-viewer.js to the `scripts` section of the `angular.json`. Before loading these files, ngx-extended-pdf-viewer checks if they're already there.

Ad 3: I've opened a bug ticket (#925).
